### ISSUE TITLE
Highlighting and selection stops working

### ISSUE THREAD

**Issue by salutis** [magit/magit#4955]
### Configuration

- macOS 13.3.1 (22E261) [for work, not my choice]
- Emacs 30 (6058b4559d4, 2022-06-05), empty `.emacs` with only `exec-path` set
- Magit 20230604.2055 [>= 3.3.0.50-git], Transient 0.4.1, Git 2.40.1, Emacs 30.0.50, darwin

### Reproduction steps

_Note:_ `toggle-debug-on-error` does not catch the error.

- open `magit-status` in a repository with 20+ untracked and 20+ unstaged files
- move the point around

[Optionally, try selecting multiple files (e.g. 10) to greatly speed up reproduction.]

**Expected:** 

Point moves [and if selecting, Magit selects the files].

**Actual:**

- custom Magit section/file highlighting stops working
- custom Magit selection stops working

[Emacs starts selecting like it does in e.g. text files, but Magit ignores that selection when e.g. staging with `s`.]

In other words, Magit becomes barely usable, not highlighting or selecting anything.

The `*Messages*` buffer has the following content, after reproduction:

```
For information about GNU Emacs and the GNU system, type C-h C-a.
Turning on magit-auto-revert-mode...done
Error in post-command-hook (magit-section-post-command-hook): (end-of-buffer)
```

The bug has been troubling me for many months now.

**Comment by salutis** [comment]
Thank you for the reply!

> One possible cause could be that another error occurred earlier, when the buffer was refreshed (i.e., its content was created). Check if there are an message that indicate so.

The `*Messages*` buffer has exactly the content I shared.  Nothing else.

> Since we cannot get a backtrace, we have to resort to debug statements. Begin with `magit-section-post-command-hook`. First comment out the call to `cursor-sensor-move-to-tangible`, a recent addition. If you can still reproduce after that, add debug statements until you locate the place where the error is triggered. It might not be enough to add debug statements to just this one function.

Augmenting `magit-section-post-command-hook`:

```emacs-lisp
(defun magit-section-post-command-hook ()
  (message "magit-section-post-command-hook-1")
  (cursor-sensor-move-to-tangible (selected-window))
  (message "magit-section-post-command-hook-2")
  (unless (bound-and-true-p transient--prefix)
    (message "magit-section-post-command-hook-3")
    (when (or magit--context-menu-buffer
              magit--context-menu-section)
      (message "magit-section-post-command-hook-4")
      (magit-menu-highlight-point-section)
      (message "magit-section-post-command-hook-5"))
    (message "magit-section-post-command-hook-6")
    (unless (memq this-command '(magit-refresh magit-refresh-all))
      (message "magit-section-post-command-hook-7")
      (magit-section-update-highlight)
      (message "magit-section-post-command-hook-8"))))
```

Messages:

```
magit-section-post-command-hook-1
magit-section-post-command-hook-2
magit-section-post-command-hook-3
magit-section-post-command-hook-6
magit-section-post-command-hook-7
Error in post-command-hook (magit-section-post-command-hook): (args-out-of-range 659424 678055)
```

This points to:

```
(magit-section-update-highlight)
```

Augmenting `magit-section-update-highlight`:

```emacs-lisp
(defun magit-section-update-highlight (&optional force)
  (message "magit-section-update-highlight-1")
  (let ((section (magit-current-section)))
    (message "magit-section-update-highlight-2")
    (when (or force
              magit-section-highlight-force-update
              (xor magit-section-pre-command-region-p (region-active-p))
              (not (eq magit-section-pre-command-section section)))
      (message "magit-section-update-highlight-3")
      (let ((inhibit-read-only t)
            (deactivate-mark nil)
            (selection (magit-region-sections)))
	(message "magit-section-update-highlight-4")
        (mapc #'delete-overlay magit-section-highlight-overlays)
	(message "magit-section-update-highlight-5")
        (setq magit-section-highlight-overlays nil)
        (setq magit-section-unhighlight-sections
              magit-section-highlighted-sections)
        (setq magit-section-highlighted-sections nil)
        (message "magit-section-update-highlight-6")
        (unless (eq section magit-root-section)
	  (message "magit-section-update-highlight-7")
          (run-hook-with-args-until-success
           'magit-section-highlight-hook section selection)
	  (message "magit-section-update-highlight-8"))
        (dolist (s magit-section-unhighlight-sections)
	  (message "magit-section-update-highlight-9")
          (run-hook-with-args-until-success
           'magit-section-unhighlight-hook s selection)
	  (message "magit-section-update-highlight-10"))
	(message "magit-section-update-highlight-11")
        (restore-buffer-modified-p nil)
	(message "magit-section-update-highlight-12")))
    (setq magit-section-highlight-force-update nil)
    (message "magit-section-update-highlight-13")
    (progn (message "magit-section-update-highlight-14")
	   (magit-section-maybe-paint-visibility-ellipses))))
```

Messages:

```
magit-section-post-command-hook-8
magit-section-post-command-hook-1
magit-section-post-command-hook-2
magit-section-post-command-hook-3
magit-section-post-command-hook-6
magit-section-post-command-hook-7
magit-section-update-highlight-1
magit-section-update-highlight-2
magit-section-update-highlight-3
magit-section-update-highlight-4
magit-section-update-highlight-5
magit-section-update-highlight-6
magit-section-update-highlight-7
Error in post-command-hook (magit-section-post-command-hook): (args-out-of-range 661651 680282)
```

This points to:

```
(run-hook-with-args-until-success
           'magit-section-highlight-hook section selection)
```

The hook has the following functions:

```
(magit-diff-highlight magit-section-highlight magit-section-highlight-selection)
```

Augmenting all three with a single `message`, we have the messages:

```
magit-section-post-command-hook-1
magit-section-post-command-hook-2
magit-section-post-command-hook-3
magit-section-post-command-hook-6
magit-section-post-command-hook-7
magit-section-update-highlight-1
magit-section-update-highlight-2
magit-section-update-highlight-3
magit-section-update-highlight-4
magit-section-update-highlight-5
magit-section-update-highlight-6
magit-section-update-highlight-7
magit-diff-highlight
Error in post-command-hook (magit-section-post-command-hook): (args-out-of-range 662858 681489)
```

This points to:

```
magit-diff-highlight
```

Augmenting `magit-diff-highlight`:

```emacs-lisp
(defun magit-diff-highlight (section selection)
  "Highlight the diff-related SECTION.
If SECTION is not a diff-related section, then do nothing and
return nil.  If SELECTION is non-nil, then it is a list of sections
selected by the region, including SECTION.  All of these sections
are highlighted."
  (message "magit-diff-highlight-1")
  (if (and (magit-section-match 'commit section)
           (oref section children))
      (progn (message "magit-diff-highlight-2")
	     (if selection
                 (dolist (section selection)
                   (magit-diff-highlight-list section selection))
               (magit-diff-highlight-list section))
             t)
    (when-let ((scope (magit-diff-scope section t)))
      (message "magit-diff-highlight-3")
      (cond ((eq scope 'region)
             (progn
	       (message "magit-diff-highlight-4")
	       (magit-diff-paint-hunk section selection t)))
            (selection
             (dolist (section selection)
	       (message "magit-diff-highlight-5")
               (magit-diff-highlight-recursive section selection)))
            (t
	     (progn
	       (message "magit-diff-highlight-6")
               (magit-diff-highlight-recursive section))))
      t)))
```

Messages:

```
magit-section-post-command-hook-8
magit-section-post-command-hook-1
magit-section-post-command-hook-2
magit-section-post-command-hook-3
magit-section-post-command-hook-6
magit-section-post-command-hook-7
magit-section-update-highlight-1
magit-section-update-highlight-2
magit-section-update-highlight-3
magit-section-update-highlight-4
magit-section-update-highlight-5
magit-section-update-highlight-6
magit-section-update-highlight-7
magit-diff-highlight-1
magit-diff-highlight-3
magit-diff-highlight-6
Error in post-command-hook (magit-section-post-command-hook): (args-out-of-range 663876 682507)
```

This points to:

```
(magit-diff-highlight-recursive section)
```

Augmenting `magit-diff-highlight-recursive`:

```emacs-lisp
(defun magit-diff-highlight-recursive (section &optional selection)
  (message "magit-diff-highlight-recursive-1")
  (pcase (magit-diff-scope section)
    ('list (message "magit-diff-highlight-recursive-2")
	   (magit-diff-highlight-list section selection))
    ('file (message "magit-diff-highlight-recursive-3")
	   (magit-diff-highlight-file section selection))
    ('hunk (message "magit-diff-highlight-recursive-4")
	   (magit-diff-highlight-heading section selection)
	   (message "magit-diff-highlight-recursive-5")
           (magit-diff-paint-hunk section selection t))
    (_     (message "magit-diff-highlight-recursive-6")
	   (magit-section-highlight section nil))))
```

Messages:

```
magit-section-post-command-hook-1
magit-section-post-command-hook-2
magit-section-post-command-hook-3
magit-section-post-command-hook-6
magit-section-post-command-hook-7
magit-section-update-highlight-1
magit-section-update-highlight-2
magit-section-update-highlight-3
magit-section-update-highlight-4
magit-section-update-highlight-5
magit-section-update-highlight-6
magit-section-update-highlight-7
magit-diff-highlight-1
magit-diff-highlight-3
magit-diff-highlight-6
magit-diff-highlight-recursive-1
magit-diff-highlight-recursive-4
magit-diff-highlight-recursive-5
```

This points to:

```
(magit-diff-paint-hunk section selection t)
```

Augmenting `magit-diff-paint-hunk`:

```emacs-lisp
(cl-defun magit-diff-paint-hunk
    (section &optional selection
             (highlight (magit-section-selected-p section selection)))
  (message "magit-diff-paint-hunk-1")
  (let (paint)
    (unless magit-diff-highlight-hunk-body
      (setq highlight nil))
    (message "magit-diff-paint-hunk-2")
    (cond (highlight
           (unless (oref section hidden)
             (add-to-list 'magit-section-highlighted-sections section)
             (cond ((memq section magit-section-unhighlight-sections)
                    (setq magit-section-unhighlight-sections
                          (delq section magit-section-unhighlight-sections)))
                   (magit-diff-highlight-hunk-body
                    (setq paint t)))))
          (t
           (cond ((and (oref section hidden)
                       (memq section magit-section-unhighlight-sections))
                  (add-to-list 'magit-section-highlighted-sections section)
                  (setq magit-section-unhighlight-sections
                        (delq section magit-section-unhighlight-sections)))
                 (t
                  (setq paint t)))))
    (message "magit-diff-paint-hunk-3")
    (when paint
      (save-excursion
	(message "magit-diff-paint-hunk-4")
        (goto-char (oref section start))
	(message "magit-diff-paint-hunk-5")
        (let ((end (oref section end))
              (merging (looking-at "@@@"))
              (diff-type (magit-diff-type))
              (stage nil)
              (tab-width (magit-diff-tab-width
                          (magit-section-parent-value section))))
	  (message "magit-diff-paint-hunk-6")
          (forward-line)
	  (message "magit-diff-paint-hunk-7")
          (while (< (point) end)
            (when (and magit-diff-hide-trailing-cr-characters
                       (char-equal ?\r (char-before (line-end-position))))
	      (message "magit-diff-paint-hunk-8")
              (put-text-property (1- (line-end-position)) (line-end-position)
                                 'invisible t))
	    (message "magit-diff-paint-hunk-9")
	    (message "put-text-property START: %s" (point))
	    (message "put-text-property END: %s" (1+ (line-end-position)))
            (put-text-property
             (point) (1+ (line-end-position)) 'font-lock-face
             (cond
              ((looking-at "^\\+\\+?\\([<=|>]\\)\\{7\\}")
	       (message "magit-diff-paint-hunk-10")
               (setq stage (pcase (list (match-string 1) highlight)
                             ('("<" nil) 'magit-diff-our)
                             ('("<"   t) 'magit-diff-our-highlight)
                             ('("|" nil) 'magit-diff-base)
                             ('("|"   t) 'magit-diff-base-highlight)
                             ('("=" nil) 'magit-diff-their)
                             ('("="   t) 'magit-diff-their-highlight)
                             ('(">" nil) nil)))
               'magit-diff-conflict-heading)
              ((looking-at (if merging "^\\(\\+\\| \\+\\)" "^\\+"))
	       (message "magit-diff-paint-hunk-11")
               (magit-diff-paint-tab merging tab-width)
               (magit-diff-paint-whitespace merging 'added diff-type)
               (or stage
                   (if highlight 'magit-diff-added-highlight 'magit-diff-added)))
              ((looking-at (if merging "^\\(-\\| -\\)" "^-"))
	       (message "magit-diff-paint-hunk-12")
               (magit-diff-paint-tab merging tab-width)
	       (message "magit-diff-paint-hunk-13")
               (magit-diff-paint-whitespace merging 'removed diff-type)
	       (message "magit-diff-paint-hunk-14")
               (if highlight 'magit-diff-removed-highlight 'magit-diff-removed)
	       (message "magit-diff-paint-hunk-15"))
              (t
               (magit-diff-paint-tab merging tab-width)
               (magit-diff-paint-whitespace merging 'context diff-type)
               (if highlight 'magit-diff-context-highlight 'magit-diff-context))))
	    (message "magit-diff-paint-hunk-16")
            (forward-line)
	    (message "magit-diff-paint-hunk-17"))))))
  (message "magit-diff-paint-hunk-16")
  (magit-diff-update-hunk-refinement section))
```

Messages:

```
magit-section-post-command-hook-1
magit-section-post-command-hook-2
magit-section-post-command-hook-3
magit-section-post-command-hook-6
magit-section-post-command-hook-7
magit-section-update-highlight-1
magit-section-update-highlight-2
magit-section-update-highlight-3
magit-section-update-highlight-4
magit-section-update-highlight-5
magit-section-update-highlight-6
magit-section-update-highlight-7
magit-diff-highlight-1
magit-diff-highlight-3
magit-diff-highlight-6
magit-diff-highlight-recursive-1
magit-diff-highlight-recursive-4
magit-diff-highlight-recursive-5
magit-diff-paint-hunk-1
magit-diff-paint-hunk-2
magit-diff-paint-hunk-3
magit-diff-paint-hunk-4
magit-diff-paint-hunk-5
magit-diff-paint-hunk-6
magit-diff-paint-hunk-7
magit-diff-paint-hunk-9
put-text-property START: 663329
put-text-property END: 663331
magit-diff-paint-hunk-16
magit-diff-paint-hunk-17
magit-diff-paint-hunk-9
put-text-property START: 663331
put-text-property END: 668163
magit-diff-paint-hunk-16
magit-diff-paint-hunk-17
magit-diff-paint-hunk-9
put-text-property START: 668163
put-text-property END: 668571
magit-diff-paint-hunk-16
magit-diff-paint-hunk-17
magit-diff-paint-hunk-9
put-text-property START: 668571
put-text-property END: 687202
magit-diff-paint-hunk-12
magit-diff-paint-hunk-13
magit-diff-paint-hunk-14
magit-diff-paint-hunk-15
Error in post-command-hook (magit-section-post-command-hook): (args-out-of-range 668571 687202)
```

This is where the error happens.

P.S. I do not always get an error; sometimes, both selection and highlighting just break, with no error message.  As to how they break: highlighting stops and selection disappears in a seemingly random fashion, as I move the point.

[All tested with an empty `.emacs`, as per the configuration detailed in the first post.]

**Comment by tarsius** [comment]
> This is where the error happens.

This *almost* gets us over the finish line. Please make it 100% certain which call to a non-magit function ultimately results in the error. I think it must be here somewhere:

```elisp
(while (< (point) end)
            (when (and magit-diff-hide-trailing-cr-characters
                       (char-equal ?\r (char-before (line-end-position))))
```

but quick experiments seem to indicate that none of these functions signal such an error when feed equivalent incorrect input.

----

This part has to be changed, or the debug statements change the outcome:

```diff
-              (if highlight 'magit-diff-removed-highlight 'magit-diff-removed)
-              (message "magit-diff-paint-hunk-15"))
+              (prog1 (if highlight 'magit-diff-removed-highlight 'magit-diff-removed)
+                (message "magit-diff-paint-hunk-15")))
```

----

```elisp
(message "put-text-property START: %s" (point))
(message "put-text-property END: %s" (1+ (line-end-position)))
put-text-property START: 668571
put-text-property END: 687202
```

This is highly surprising.

- The current position is by definition somewhere within the current line.
- The end of a line is the last position within that line — a tautology.

But here, somehow, the current position comes hundreds of characters after the end of the line it is part of.

----

Before you do anything else, I would recommend reinstalling Magit: Uninstall Magit, restart Emacs, reinstall Magit. Often when impossible bugs happen, the cause is that an old, already loaded version of a package has leaked into the byte-code of a new version. Here I have my doubts, but it is still worth trying.

- Navigate to the reported positions, maybe you notice something interesting there.
- Does bidirectional text appear in this buffer?
- Try replacing `line-end-position` with `pos-eol`. Do they report the same value?
- Additionally printing `point-max` might also be interesting.
- Call `widen` just in case.

**Comment by salutis** [comment]
> Please make it 100% certain which call to a non-magit function ultimately results in the error.

It seems to me (based on the output I shared) that the error happens in `put-text-property`.

> Before you do anything else, I would recommend reinstalling Magit: Uninstall Magit, restart Emacs, reinstall Magit. Often when impossible bugs happen, the cause is that an old, already loaded version of a package has leaked into the byte-code of a new version. Here I have my doubts, but it is still worth trying.

Done.

To brand new Magit, I added the following six `message` calls:

```emacs-lisp
(cl-defun magit-diff-paint-hunk
    (section &optional selection
             (highlight (magit-section-selected-p section selection)))
  (let (paint)
    (unless magit-diff-highlight-hunk-body
      (setq highlight nil))
    (cond (highlight
           (unless (oref section hidden)
             (add-to-list 'magit-section-highlighted-sections section)
             (cond ((memq section magit-section-unhighlight-sections)
                    (setq magit-section-unhighlight-sections
                          (delq section magit-section-unhighlight-sections)))
                   (magit-diff-highlight-hunk-body
                    (setq paint t)))))
          (t
           (cond ((and (oref section hidden)
                       (memq section magit-section-unhighlight-sections))
                  (add-to-list 'magit-section-highlighted-sections section)
                  (setq magit-section-unhighlight-sections
                        (delq section magit-section-unhighlight-sections)))
                 (t
                  (setq paint t)))))
    (when paint
      (save-excursion
        (goto-char (oref section start))
        (let ((end (oref section end))
              (merging (looking-at "@@@"))
              (diff-type (magit-diff-type))
              (stage nil)
              (tab-width (magit-diff-tab-width
                          (magit-section-parent-value section))))
          (forward-line)
          (while (< (point) end)
            (when (and magit-diff-hide-trailing-cr-characters
                       (char-equal ?\r (char-before (line-end-position))))
              (put-text-property (1- (line-end-position)) (line-end-position)
                                 'invisible t))

	    (message "point-max: %s" (point-max))
	    (message "end: %s" (marker-position end))
	    
	    (message "will put-text-property START: %s" (point))
	    (message "will put-text-property END: %s" (1+ (line-end-position)))
	    
            (put-text-property
             (point) (1+ (line-end-position)) 'font-lock-face
             (cond
              ((looking-at "^\\+\\+?\\([<=|>]\\)\\{7\\}")
               (setq stage (pcase (list (match-string 1) highlight)
                             (`("<" nil) 'magit-diff-our)
                             (`("<"   t) 'magit-diff-our-highlight)
                             (`("|" nil) 'magit-diff-base)
                             (`("|"   t) 'magit-diff-base-highlight)
                             (`("=" nil) 'magit-diff-their)
                             (`("="   t) 'magit-diff-their-highlight)
                             (`(">" nil) nil)))
               'magit-diff-conflict-heading)
              ((looking-at (if merging "^\\(\\+\\| \\+\\)" "^\\+"))
               (magit-diff-paint-tab merging tab-width)
               (magit-diff-paint-whitespace merging 'added diff-type)
               (or stage
                   (if highlight 'magit-diff-added-highlight 'magit-diff-added)))
              ((looking-at (if merging "^\\(-\\| -\\)" "^-"))
               (magit-diff-paint-tab merging tab-width)
               (magit-diff-paint-whitespace merging 'removed diff-type)
               (if highlight 'magit-diff-removed-highlight 'magit-diff-removed))
              (t
               (magit-diff-paint-tab merging tab-width)
               (magit-diff-paint-whitespace merging 'context diff-type)
               (if highlight 'magit-diff-context-highlight 'magit-diff-context))))

	    (message "did put-text-property START: %s" (point))
	    (message "did put-text-property END: %s" (1+ (line-end-position)))
	    
            (forward-line))))))
  (magit-diff-update-hunk-refinement section))
```

Triggering the problem results in the following messages:

```
[...]
point-max: 287692
end: 1153265
will put-text-property START: 286892
will put-text-property END: 287245
did put-text-property START: 286892
did put-text-property END: 287245
point-max: 287692
end: 1153265
will put-text-property START: 287245
will put-text-property END: 287690
did put-text-property START: 287245
did put-text-property END: 287690
point-max: 287692
end: 1153265
will put-text-property START: 287690
will put-text-property END: 287693
Error in post-command-hook (magit-section-update-highlight): (args-out-of-range 287690 287693)
```

Notice how `put-text-property END: 287683` is one over `point-max: 287692`.  Is that normal?

>     * Navigate to the reported positions, maybe you notice something interesting there.

I have checked 3 different locations (based on selecting differently) and found nothing special, only plain ASCII `+`/`-` hunks.

>     * Does bidirectional text appear in this buffer?

Not that I know of.

>     * Try replacing `line-end-position` with `pos-eol`. Do they report the same value?

The same.

>     * Additionally printing `point-max` might also be interesting.

See the log above.

>     * Call `widen` just in case.

I executed `M-x widen RET` in the `magit-status` buffer before re-testing.  No difference.

**Comment by salutis** [comment]
I found that it might be the `emacs.history` file in my Git repository that is breaking Magit's highlighting and selection.

@tarsius Could you please add the attached file to a repository to see if it breaks Magit highlighting and selection for you?  (For greatest reproduction probability, put the file into a repository with a large number of not-yet-committed files and changes.)

[emacs.history.zip](https://github.com/magit/magit/files/11732937/emacs.history.zip)

**Comment by tarsius** [comment]
> Notice how `put-text-property END: 287683` is one over `point-max: 287692`.  Is that normal?

Sounds like a off-by-one error, which we could counter by throwing a `(min (point-max) ...)` in there.

But this is not actually "one over"; it is "nine below".

I'm a bit busy with other things and I likely won't respond to the rest for a few days.

**Comment by salutis** [comment]
> But this is not actually "one over"; it is "nine below".

Oops!  I cannot read.  😄 

P.S. I am increasingly certain the `emacs.history` file attached above causes the problem.

**Comment by salutis** [comment]
_Update:_

In desperation, I deleted my `emacs.history` file, but Magit struggles with the new one too.  Maybe the problem is caused by the `savehist-mode` auto-saving the history (on its timer}?

**Comment by salutis** [comment]
I disabled the `savehist-mode` and Magit still struggles.  So, it is probably something else.  Below, I am attaching the new history file which, upon automatic changes made by Emacs, triggers the problem.

[emacs.history.zip](https://github.com/magit/magit/files/11951725/emacs.history.zip)

**Comment by tarsius** [comment]
@salutis I have deleted your last message because it contained a link to a file that most likely contains sensitive information.

Here is that message, without the link:

> I disabled the `savehist-mode` and Magit still struggles.  So, it is probably something else. Below, I am attaching the new history file which, upon automatic changes made by Emacs, triggers the problem.
> 
> emacs.history.zip

**Comment by tarsius** [comment]
Your file is huge and contains lines that are longer than 50000 characters (i.e., more than `long-line-threshold`). There is a new feature in Emacs that narrows the buffer during `post-command-hook` (and others), when long lines are detected.

This cannot be counteracted using `widen`. But this would work:

```diff
diff --git a/lisp/magit-section.el b/lisp/magit-section.el
index 03883475bb..4dacbac513 100644
--- a/lisp/magit-section.el
+++ b/lisp/magit-section.el
@@ -1588,7 +1588,9 @@ (defun magit-section-post-command-hook ()
               magit--context-menu-section)
       (magit-menu-highlight-point-section))
     (unless (memq this-command '(magit-refresh magit-refresh-all))
-      (magit-section-update-highlight))))
+      (without-restriction
+        :label 'long-line-optimizations-in-command-hooks
+        (magit-section-update-highlight)))))
 
 (defun magit-section-deactivate-mark ()
   (setq magit-section-highlight-force-update t))
```

However I plan to play along with the intention of this new feature and instead disable section highlighting. So the effect would be basically the same as now, we would just handle it more gracefully, by informing the user that this has happened.

> - custom Magit selection stops working

It still works (commands that act on the "section selection" still use it), the selection "just" isn't visualized anymore, which of course makes it harder to use it.

**Comment by salutis** [comment]
> Your file is huge and contains lines that are longer than 50000 character [...]

Just when I thought that Emacs can _finally_ handle long lines. Nope! :)

P.S. I commented on https://github.com/magit/magit/commit/986515d93f2214895e73e1066e9101b1d371fb62.

Thank you.

**Comment by rudolf-adamkovic** [comment]
_Note:_ Yet another solution is to write `emacs.history binary` to `.gitattributes`.
